compiler/natives/os/os.go

```diff
@@ -24,3 +24,27 @@ func init() {
 }
 
 func runtime_beforeExit() {}
+
+// Override the builtin os.Hostname(). The builtin Hostname() will result in
+// a non-working syscall on most (all?) platforms
+func Hostname() (string, error) {
+	// Browser
+	// The browser doesn't know anything about any hostname, so we use the host part of
+	// the current url.
+	if location := js.Global.Get("location"); location != js.Undefined {
+		if browserHostname := location.Get("hostname"); browserHostname != js.Undefined {
+			return browserHostname.String(), nil
+		}
+	}
+
+	// Node.js
+	// We call node's require("os").hostname() which should return current hostname
+	if require := js.Global.Get("require"); require != js.Undefined {
+		// "os" is in nodejs core, it should always exist
+		if nodeHostname := require.Invoke("os").Call("hostname"); nodeHostname != js.Undefined {
```

Comment:
You're right, it would be perhaps an odd failure case in a browser. 

I think I was assuming (without closely reading your code) something more sophisticated than window.location, as that does not seem like the right host name to be returning. If the client host name cannot be determined, would it not make more sense to return '' or perhaps 'localhost' in the browser? 

On January 30, 2016 1:01:21 PM GMT+01:00, Anders Brander <notifications@github.com> wrote:
>> +// a non-working syscall on most (all?) platforms
>> +func Hostname() (string, error) {
>> +	// Browser
>> +	// The browser doesn't know anything about any hostname, so we use
>the host part of
>> +	// the current url.
>> +	if location := js.Global.Get("location"); location != js.Undefined
>{
>> +		if browserHostname := location.Get("hostname"); browserHostname !=
>js.Undefined {
>> +			return browserHostname.String(), nil
>> +		}
>> +	}
>> +
>> +	// Node.js
>> +	// We call node's require("os").hostname() which should return
>current hostname
>> +	if require := js.Global.Get("require"); require != js.Undefined {
>> +		// "os" is in nodejs core, it should always exist
>> +		if nodeHostname := require.Invoke("os").Call("hostname");
>nodeHostname != js.Undefined {
>
>For that to happen a few planets have to be aligned:
>
>- The browser doesn't implement the global `location` object. Is there
>any browsers that doesn't?
>- The user uses browserify.
>- The user expects `os.Hostname()` to not fail.
>
>After this PR the result would be a catastrophic Javascript failure.
>Before this PR the result would be a catastrophic gopherjs failure.
>
>I think we can safely ignore this corner case, but I'm open for
>suggestions.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/gopherjs/gopherjs/pull/385/files#r51344206
>
>!DSPAM:56aca6ec24651115619531!

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.